textdb/textdb-dataflow/src/main/java/edu/uci/ics/textdb/dataflow/common/KeywordPredicate.java

```diff
@@ -52,23 +59,48 @@ public KeywordPredicate(String query, List<Attribute> attributeList, Analyzer an
     }
 
     /**
-     * Creates a Query object as a boolean Query on all attributes.
+     * Creates a Query object as a boolean Query on all attributes
      * Example: For creating a query like
      * (TestConstants.DESCRIPTION + ":lin" + " AND " + TestConstants.LAST_NAME + ":lin")
      * we provide a list of AttributeFields (Description, Last_name) to search on and a query string (lin)
      *
      * TODO #88:BooleanQuery() is deprecated. In future a better solution could be worked out in Query builder layer
-
-     * @return QueryObject
+     *
+     * @return Query
      * @throws ParseException
      */
     private Query createQueryObject() throws ParseException {
+        ArrayList<String> tokens;
+        List<String> fieldList = new ArrayList<String>();
         BooleanQuery booleanQuery = new BooleanQuery();
-        MultiFieldQueryParser parser = new MultiFieldQueryParser(this.fields, this.analyzer);
-        for(String searchToken: this.tokens){
+
+        for(int i=0; i< attributeList.size(); i++){
+            String fieldName = attributeList.get(i).getFieldName();
+            /*
+            If the field is of type string, handle it differently
+             */
+            if(attributeList.get(i).getFieldType() == FieldType.DATE.STRING){
+                Query termQuery = new TermQuery(new Term(fieldName,query));
+                booleanQuery.add(termQuery, BooleanClause.Occur.SHOULD);
+            }
+            else {
+                fieldList.add(fieldName);
+            }
+        }
+
+        if(fieldList.size()==0){
+            return booleanQuery;
+        }
+
+        String[] remainingFields = (String[]) fieldList.toArray(new String[0]);
```

Comment:
I am not quite following this logic.  Why do we need to treat `remainingFields` differently?  Why cannot we use a single loop to handle all the fields?  Also comments will help.